extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridConverterBase.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    internal abstract class EventGridConverterBase  : IInputConverter
+    {
+        public EventGridConverterBase()
+        {
+        }
+
+        public async ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            try
+            {
+                if (context is null)
+                {
+                    throw new ArgumentNullException(nameof(context));
+                }
+
+                if (context.Source is not string json)
+                {
+                    throw new InvalidOperationException($"Context source must be a non-null string. Current type of context source is '{context.Source.GetType()}'");
+                }
+
+                var result = ConvertCoreAsync(context.TargetType, json);
```

Comment:
The implementations you have are returning `ConversionResult` from their `ConvertCoreAsync` implementation. This is then wrapping it in a success result. This doesn't seem right to me, should there be a type check here?

Or maybe it would be simpler to have `ConvertCoreAsync` just always return `ConversionResult` - let the implementations perform the wrapping of `ConversionResult.Success()`

Reply:
Ahh whoops, thanks for catching that, definitely wasn't the intension. Going with the latter suggestion